### PR TITLE
implement asin and acos functions, ref: #2109

### PR DESCRIPTION
**Pull request by lakshayg**

### CODE REVIEW

File: Source/Math/TensorOps.h [hunk 1]
```diff
@@ -316,6 +320,8 @@ DefBinaryOp(ElementwiseProductWithReciprocalDerivative, a * -Sqr(b)); // b = out
 DefBinaryOp(ElementwiseProductWithSqrtDerivative, a / (2 * b)); // b = output; d/dx sqrt(x) = 1/(2 * sqrt(x)) --> note this is the same as ElementwiseQuotient w a constant; if more show up like this we should add more template params
 DefBinaryOp(SqrOfDifference, Sqr(a - b));
 DefBinaryOp(ElementwiseProductWithExponentialLinearUnitDerivativeFromOutput, b >= 0 ? a : a*(1+b)); // b = output;
+DefBinaryOp(ElementwiseProductWithAsinDerivative, a / sqrt_(1 - b * b)); // note: b = input for asin()
+DefBinaryOp(ElementwiseProductWithAcosDerivative, -a / sqrt_(1 - b * b)); // note: b = input for acos()
```
Comment (spandantiwari) [inline Source/Math/TensorOps.h]: No action needed. You don't need update this. I will update this on our side.

File: bindings/python/cntk/ops/__init__.py [hunk 2]
```diff
@@ -1472,6 +1472,50 @@ def cos(x, name=''):
     return cos(x, name)
 
 @typemap
+def acos(x, name=''):
+    '''
+    Computes the element-wise acos of ``x``:
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: If you agree, no action is needed. I will update this on our side.

File: bindings/python/cntk/ops/__init__.py [hunk 3]
```diff
@@ -1472,6 +1472,50 @@ def cos(x, name=''):
...
+        :class:`~cntk.ops.functions.Function`
+    '''
+    from cntk.cntk_py import acos
+    x = sanitize_input(x)
+    return acos(x, name)
+
+@typemap
+def asin(x, name=''):
+    '''
+    Computes the element-wise asin of ``x``:
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: No action needed. You don't need update this. I will update this on our side.

**Comment by lakshayg** [comment]
@spandantiwari All the changes you proposed look good. Please go ahead.

**Comment by spandantiwari** [comment]
This is now in master with commit e79c27688ab56f8126a1ef88ed792a08d12bfa33.

**Pull request closed without merging**